nixos/modules/programs/steam.nix

```
@@ -0,0 +1,28 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.steam;
+in {
+  options.programs.steam = {
+    enable = mkEnableOption "steam";
+  };
+
+  config = mkIf cfg.enable {
+    hardware.opengl = { # glXchooseVisual failed https://github.com/NixOS/nixpkgs/issues/47932
+      enable = true;
+      driSupport32Bit = true;
+    };
+
+    # optionally enable 32bit pulseaudio support if pulseaudio is enabled
+    hardware.pulseaudio.support32Bit = true;
```

```suggestion
    hardware.pulseaudio.support32Bit = config.hardware.pulseaudio.enable;
```

Most people will probably have it enabled anyways if they use steam but this would be more consistent

---

Alternatively, you could just enable pulseaudio too. Is there any reason why you might not want pulse if you use Steam?

---

I don't see any. Some people insist on ALSA, that might be the only one I can think of.